src/math_utils.rs

```diff
@@ -40,6 +40,21 @@ pub fn safe_div(x: &BigInt, y: &BigInt) -> Result<BigInt, VirtualMachineError> {
     Ok(q)
 }
 
+/// Performs integer division between x and y; fails if x is not divisible by y.
+pub fn safe_div_usize(x: usize, y: usize) -> Result<usize, VirtualMachineError> {
+    if y.is_zero() {
+        return Err(VirtualMachineError::DividedByZero);
+    }
+
+    let (q, r) = x.div_mod_floor(&y);
+
+    if !r.is_zero() {
+        return Err(VirtualMachineError::SafeDivFailUsize(x, y));
+    }
+
+    Ok(q)
+}
+
```

Comment:
Let's implement it as [this trait](https://docs.rs/num-traits/latest/num_traits/ops/checked/trait.CheckedDiv.html) instead.

Reply:
`checked_div()` is a different operation since [it'll succeed](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3867b442742dd31ceed34d9f6aac2145) when the remainder is not zero.